src/Runtime/jni/jniwrapper.c

```diff
@@ -504,6 +504,10 @@ OMTensorList *omtl_java_to_native(
       omTensorListCreate(jni_omts, (int64_t)jomtl_omtn), jni_omtl != NULL, env,
```

Comment:
Can you remove the comment "Note that we do own..."? It no longer applies since now omTensorListCreate copies jni_omts.

Also, you need to replace `omTensorCreate` with `omTensorCreateWithOwnership(owning=1)` to free the input tensors themselves.

Reply:
I can remove the comment but the `omTensorCreate` is still valid. input tensors' data is still owned by Java (or to be more specific, the caller), so we shouldn't mark the tensor as needing to free the data.

Reply:
Ah yes indeed...mind glitch.